USER_MANUAL.md

```diff
@@ -537,6 +588,24 @@ automatically. It will also automatically sort associative-commutative
 collections, and flatten the indentation of associative collections, when
 unparsing.
 
+### `public` and `private` attribute
+
+K allows users to declare certain pieces of syntax as either public or private.
+All syntax is public by default. Public syntax can be used from any module that
+imports that piece of syntax. A piece of syntax can be declared private with
+the `private` attribute. This means that that syntax can only be used in the
+module in which it is declared; it is not visible from modules that import
+that module.
+
+You can also change the default visibility of a module with the `private`
+attribute, when it is placed directly on a module. A module with the `private`
+attribute has all syntax `private` by default; this can be overridden on
+specific sentences with the `public` attribute.
+
+Note that the `private` module attribute also changes the default visiblity
+of imports; please refer to the appropriate section elsewhere in the manual
+for more details.
+
 
```

Comment:
I'm not sure how to suggest code blocks, but I think we should add an example.

```suggestion
We give an example of possible usage below:

module WIDGET
  syntax Widget ::= gadget()
endmodule

module FOO [private]
  imports WIDGET                         // <- B may not use `gadget()`
  syntax Foo ::= foo()                   // <- B may not use this symbol
  syntax Foo ::= bam() [public]          // <- B may use this symbol
  syntax Foo ::= fu() [private]          // <- B may not use this symbol
  syntax Foo ::= "bar" [public, token]   // <- B may use this token
               | r"ba." [token, prec(3)] // <- B may not use this token
endmodule

module B
  imports private FOO                    // <- B may use the _public_ syntax of A, but C may not
  syntax Bar ::= a() [public]            // <- C may use this symbol
               | b() [private]           // <- C may not use this symbol
endmodule

module C
  imports B
  imports WIDGET                         // C may now use WIDGET's public interface
endmodule

```

Reply:
I agree that we should add an example like this.